app/src/full/kotlin/io/homeassistant/companion/android/location/LocationProvider.kt

```diff
@@ -0,0 +1,19 @@
+package io.homeassistant.companion.android.location
+
+import android.Manifest
+import android.content.Context
+import android.location.Location
+import androidx.annotation.RequiresPermission
+import com.google.android.gms.location.LocationServices
+import kotlinx.coroutines.tasks.await
+import timber.log.Timber
+
+@RequiresPermission(allOf = [Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION])
+suspend fun getLocation(context: Context): Location? {
+    return try {
+        LocationServices.getFusedLocationProviderClient(context).lastLocation.await()
```

Comment:
I understand why you're returning the last location here, as that is what the code you're replacing is doing, but the function name suggests to me I'm getting the current location.

Reply:
renamed to getLastLocation